noir-projects/noir-contracts/contracts/token_contract/src/main.nr

```diff
@@ -448,15 +451,18 @@ contract Token {
         // TODO(#7775): Manually fetching the randomness here is not great. If we decide to include randomness in all
         // notes we could just inject it in macros.
 
-        /// Safety: We use the randomness to preserve the privacy of the note recipient by preventing brute-forcing,
-        /// so a malicious sender could use non-random values to make the note less private. But they already know
-        /// the full note pre-image anyway, and so the recipient already trusts them to not disclose this
-        /// information. We can therefore assume that the sender will cooperate in the random value generation.
-        let note_randomness = unsafe { random() };
-        let note_setup_payload = UintNote::setup_payload().new(to, note_randomness, to_note_slot);
+        // Safety: We use the randomness to preserve the privacy of the note recipient by preventing brute-forcing,
+        // so a malicious sender could use non-random values to make the note less private. But they already know
+        // the full note pre-image anyway, and so the recipient already trusts them to not disclose this
+        // information. We can therefore assume that the sender will cooperate in the random value generation.
+
+        let partial = PartialUintNote::new(to, to_note_slot);
+        let encrypted_log = compute_log(*context, partial, to, from);
+        context.emit_private_log(encrypted_log);
```

Comment:
Yeah, as discussed earlier, this will look neater, and avoid touching low-level functions if you use events instead.
Something roughly like:

```noir
let partial_note_event = PartialNoteEvent::new(...);
partial_note_event.emit(encode_and_encrypt_event(&mut context, to, from));

// where encode_and_encrypt_event is imported from the log assembly strategies dir
```